src/main.rs

```diff
@@ -16,8 +17,57 @@ use localauth0::{controller, APP_NAME};
 // Singleton logger. Used to free user from manually passing Logger objects around.
 static LOGGER_GUARD: GuardLoggerCell = GuardLoggerCell::new();
 
+fn main() -> Result<(), Box<dyn Error>> {
+    match std::env::args().nth(1).as_deref() {
+        Some("healthcheck") => Ok(healthcheck()?),
+        _ => Ok(server()?),
+    }
+}
+
+async fn is_endpoint_healthy(client: &reqwest::Client, endpoint: String) -> bool {
+    let status = client.get(&endpoint).send().await.map(|r| r.error_for_status());
+
+    match status {
+        Ok(Ok(_)) => {
+            println!("{endpoint} OK");
+            true
+        }
+        _ => {
+            println!("{endpoint} ERROR {status:?}");
+            false
+        }
+    }
+}
+
+fn healthcheck() -> Result<(), String> {
+    tokio::runtime::Builder::new_current_thread()
+        .enable_all()
+        .build()
+        .unwrap()
+        .block_on(async {
+            let config = Config::load_or_default();
+
+            let client = reqwest::Client::builder()
+                .danger_accept_invalid_certs(true)
+                .build()
+                .unwrap();
+
+            let endpoint_http = format!("http://127.0.0.1:{}/healthcheck", config.http().port());
```

Comment:
i think that `/check` might be more conforming to primait standards

Reply:
Done